govcd/generic_functions.go

```diff
@@ -36,3 +42,34 @@ func oneOrError[T any](key, name string, entitySlice []*T) (*T, error) {
 
 	return entitySlice[0], nil
 }
+
+// readFileAndUnmarshalJSON reads a file and unmarshals it to the given variable
+func readFileAndUnmarshalJSON[T any](filename string, object T) error {
+	data, err := os.ReadFile(path.Clean(filename))
+	if err != nil {
+		return fmt.Errorf("failed to read from file: %s", err)
+	}
+
+	err = json.Unmarshal(data, object)
+	if err != nil {
+		return fmt.Errorf("failed to unmarshal file contents to the object: %s", err)
+	}
+
+	return nil
+}
+
+// marshalJSONAndWriteToFile marshalls the given object into JSON and writes
+// to a file with the given permissions in octal format (e.g 0600)
+func marshalJSONAndWriteToFile[T any](filename string, object *T, permissions int) error {
```

Comment:
Same observation given for `readFileAndUnmarshalJSON`. I don't see the need for using generics